### ISSUE TITLE
Review curriculum and curriculum update process, charter cross-campus curriculum committee

### ISSUE THREAD

**Issue by rebeccaradding** [foundersandcoders/master-reference#735]
# Review curriculum and curriculum update process, charter cross-campus curriculum committee

The curriculum working group is in the process of trialing improvements to the process of updating the curriculum. Thus far, the beginnings of this process has been documented in various issues such as #651, as well as in [`curriculum-update-process.md`](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/curriculum-update-process.md).

I am raising this issue to encourage further improvements to this process, and to move us as an international community toward clarity around coordinating curriculum development across three campuses. 

### Conduct review of curriculum update process

We need to conduct a review of "week facilitator" structure employed in most recent curriculum update process. 

_If we continue with week facilitators, we should:_
- Sort out week facilitators for the next cycle and possibly longer @sofer 
- Create documentation describing the responsiblities of a week facilitator 
- Clarify whether week facilitators are maintainers of their weeks resources (some resources are still external)

_If we choose an alternative structure for making updates_, we will need to produce clear documentation to guide the next cycle accordingly.

Based on the results of the discussion above, we need to revise [`curriculum-update-process.md`](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/curriculum-update-process.md) so that it describes the general update process (taking into account learnings from this cycle) and provides a timeline that is week, but not date-specific (*note:* this schedule must account for staggered schedules across campuses - week N in London is Week N-2 in Gaza)

We also need to define and document common standards for resources such as workshops and morning challenges (for example: solution in folder, not branch).

### Conduct curriculum review

- Update the curriculum vision #591 to ensure that that how we design and organise resources aligns with the intended [self-organising nature of the programme](https://github.com/foundersandcoders/master-reference/blob/master/self-organising.md)
- If appropriate, draft additional documentation with philosophical guidance on the curriculum #591
- Review scope of learning outcomes weeks 1-8 #691 @sofer 
- Review learning outcomes weeks 10-16 #692 @sofer

### Cross-campus curriculum committee

- Draft a curriculum committee charter #599 so that all programmes clearly understand which decision-making powers have been delegated 
- Ensure participation by all campuses on this committee, in particular by reaching out to Gaza graduates and setting a meeting schedule that enables their participation (particularly relevant for Gaza)

### General

- Identify long-term maintainers of the `master-reference` (we have removed this from the job description of course facilitator) @sofer 
- Define the responsibilities of a `master-reference` maintainer #370 
- Update CONTRIBUTING.md #693 to reflect any changes to process that come out of these discussions

## Action steps

- Call for additional participants in the curriculum working group (see the [`register`](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/working-group/members.md) for current curriculum participants)
- Convene a working group meeting to conduct reviews and divide up tasks

**Comment by rebeccaradding** [comment]
Doodle for working group meeting to tackle the first action step - conduct review of curriculum update process: https://doodle.com/poll/bqhxt4r9d49xeh8b

At this meeting, I suggest we review the other proposed action items to decide on a schedule and division of labor.

If you were a week facilitator, _but aren't interested in continuing in this role,_ it would be helpful if you could attend this review meeting, and let us know you're stepping down after providing feedback on process.

@finnhodgkin @ZooeyMiller @sofer @des-des @yvonne-liu @mantagen @shiryz @yvonne-liu @ronanyeah @sohilpandya @m4v15 @mattlub @RanaAlqrenawi @jsms90 @eliascodes @oliverjam @astroash

**Comment by eliascodes** [comment]
Suggest we do this Wednesday lunchtime since it has join-most attendees and representation from each campus (Gaza: Me/Ronan, Nazareth: Shireen/Mavis/Eoin, London: Me/Eoin/Ronan(?)).

**Comment by rebeccaradding** [comment]
@eliascodes thanks for reviewing the results, @sofer hasn't filled the doodle but it appears he can attend at that time as well (important for continunity's sake). I'll get it on the calendar and send out invites.

**Comment by rebeccaradding** [comment]
Scheduled for Wednesday, November 15 at 13:00 London time, 15:00 Gaza/Nazareth time. Hangout in calendar!

**Comment by rebeccaradding** [comment]
Agenda for today: https://hackmd.io/AwdhFYA4BN3BaSwDGJ4BYBGk2YIYBme8AnCAciQEzgjIBsIeQA==

**Comment by rebeccaradding** [comment]
### Action step:

On Friday, November 17 at 13:00 London (event on FAC calendar, link in Hangout), @sofer will meet with @eliascodes @des-des @ronanyeah @m4v15 and any other interested parties to prepare recommendations on the following issues:

- A curriculum style guide
- A description of the role of the WF
- A description of the role of the CF in the curriculum update process
- Various issues related to `master-reference` maintenance

The wider group is invited to reconvene on Friday, November 24 at 13:00 (event on calendar, link in Hangout) to review the recommendations and address the next set of action steps.

**Comment by rebeccaradding** [comment]
@sofer @eliascodes @des-des @ronanyeah @m4v15 what were the results of Friday's meeting? Looking forward to your recommendations!

**Comment by m4v15** [comment]
@yvonne-liu wrote some minutes I think - actually there might be some confusion on who was supposed to submit the final PR or whatever, can see them [here](https://hackmd.io/s/HktxQaK1M) tho.

**Comment by eliascodes** [comment]
@rebeccaradding there's also [this summary](https://hackmd.io/JwZjEMFYAZgWgBwDYAmAWObgJYtJw5IEQBTBAJgHZJTwBGYIA===#) written by @m4v15.

**Comment by m4v15** [comment]
Yeah that summary was a bit rough, but now I've kind of formalised it up into something that could be a proposal/something to discuss on Friday

**Comment by eliascodes** [comment]
Anyone coming to tomorrows meeting, would be good if you could have a quick read through Mavis' summary/proposal I linked before. :+1:

**Comment by m4v15** [comment]
Just to clarify: 

**[Initial minutes, with overview of general consensus](https://hackmd.io/s/HktxQaK1M)**

**[My more detail possible implementation of those ideas with regards to the course](https://hackmd.io/JwZjEMFYAZgWgBwDYAmAWObgJYtJw5IEQBTBAJgHZJTwBGYIA===#)**

**Comment by m4v15** [comment]
# Action Points of meeting today

- [ ] Make the proposal for London meeting @m4v15 @eliascodes
- [ ] Update CF job description @sofer
- [ ] Learning outcomes of MR @eliascodes
- [ ] Identify someone to write style guide (community)
- [ ] Identify potential MR maintainers

**Comment by astroash** [comment]
I'm happy to put together a style guide! I just read through a bunch of them and it made me sad 😅

**Comment by astroash** [comment]
In regards to the reference above, is the style guide for the creation of weekly learning outcomes or for all content in master reference?

**Comment by eliascodes** [comment]
@astroash my understanding was that it was a style guide on the creation and maintenance of curriculum resources.

**Comment by astroash** [comment]
Also thinking about how these points would fall into the flow of the above implementation:

### Who reviews and closes issues on master reference with week labels?
Currently there are a number of issues on maz ref with feedback/changes for the week made by mentors/CFs/current students from each campus. These issues will need to be looked at before decisions on week changes are made by MRM. This could be solved by either:
- MRM's being responsible for gathering info from these and closing ones they think are unneeded
- All campus CF's meeting to discuss which ones they think are relevant and feed info back to MRM's in combination with students feedback
- Individual cohorts to look through all issues together and feedback which changes they think are needed (task repeated in all 3 campuses)
- Individual cohorts to look through issues raised in their own campus and feedback which changes they think are needed

I don't have a preference for any of the options (or any other solutions), but thought it's worth highlight who's remit this falls under `¯\_(ツ)_/¯`

### How does Gaza and Naz campuses affect student working groups for editing weeks?
Currently students from Gaza and Nazareth aren't necessarily mentoring a week for the next cohort so I was wondering if/how this would impact the effectiveness of student working groups for editing weeks? This is more of a question for current/past facilitators of these campuses on how they imagine this working.

**Comment by astroash** [comment]
> my understanding was that it was a style guide on the creation and maintenance of curriculum resources.

@eliascodes, makes sense. So it will be an extension of the current [`creation-of-resources.md`](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/creation-of-resources.md)

**Comment by eliascodes** [comment]
> Currently students from Gaza and Nazareth aren't necessarily mentoring a week for the next cohort

News to me. Where'd this come from?

> Who reviews and closes issues on master reference with week labels?

Currently, MRM.

In future, it mostly depends on what the feedback process looks like. IMO it would be easiest for MRM to be handed one consolidated set of feedback for each cohort/campus.

I mentioned in the meeting that the CF should be able to design the feedback process so it's actually effective for their particular cohort. If part of that process is to have them opening issues during the curriculum, then students should review and close those issues (their own cohort's) when doing their final feedback session in week 10/11/whatever.

I imagine MRM will perform periodic sweeps of (un)dead issues anyway though.

**Comment by ZooeyMiller** [comment]
> > Currently students from Gaza and Nazareth aren't necessarily mentoring a week for the next cohort
> 
> News to me. Where'd this come from?

There are, AFAIK, only 2 students from G2 not mentoring at all, though a couple have taken a back seat. AFAIK all of G3 will be mentoring G4.

I think the not mentoring the next cohort thing is more a Nazareth problem than a Gaza problem.

**Comment by m4v15** [comment]
@astroash 

>Currently students from Gaza and Nazareth aren't necessarily mentoring a week for the next cohort so I was wondering if/how this would impact the effectiveness of student working groups for editing weeks? This is more of a question for current/past facilitators of these campuses on how they imagine this working.

yeah like @ZooeyMiller says this is much more a Nazareth issue than anything else, and tbh I don't think this impact's the effectiveness of WWG making edits, it seems to have more of an impact in terms of delivery of the next course - in London/Gaza because the students who are delivering the week have been working on it for a while they are well prepared. In Nazareth it just means whoever does end up delivering it has to do more work to be well prepared and know's about any changes. The idea is that MRM's are aware of all big changes, so can feed this back to CF's who would feed it back to the mentors (in Nazareth, or wherever else this is needed)

**Comment by astroash** [comment]
Thanks for all the answers 😸, they cleared up a few things I was unsure about. I've created a _super rough_ style guide for master reference to get peoples feedback on. There's a couple of things I added in as I thought they made sense, and a few things I was unsure about so I have noted them in all the **Extra Stuff** section at the bottom. Let me know what you all think. [Scrutinise Me](https://hackmd.io/IYVgpgzAHAnFwFoCMMQDYEBYoBMDGCARumAjgGaEBMOADCLTIXlEA===?view)

**Comment by eliascodes** [comment]
First draft of "proposal" here: https://hackmd.io/BwThCYGMBNoBgLQEYmWAgLAdhJBYA2JZAgIwENS4DgoQ4g==#

Comments welcome and encouraged. It is unavoidably quite long 😞 

This will be brought to the London business meeting, framed as an FYI rather than something to be voted on.

**Comment by astroash** [comment]
This doc is sweet 🍬 🍬 🍬 Digging the diagram! I have a few questions...

>### Master-reference Maintainers
>
>Immediate term
>- Review and update weekly learning outcomes as a result (2)

Are you considering reviewing the weekly outcomes before week 10? As we're coming up to the first implementation of this structure it might be worth waiting until post CF feedback to avoid duplicated work.

>Regularly
>
>- Review & merge pull requests to MR and all curriculum resources ensuring the quality and coherence of the resources

I know this is flagged up in the cons, but this sounds like a huge time commitment.

> ### Rough idea, adaptable to needs of each campus/CF
>
>Week 10
>
> **CF** - Find time for students to do this (3-6 hours). Raise an issue on the MR for the MRM, with problems and proposed solutions (if possible)

Is the idea that each campus makes an epic issue on everything they want changed? Do campuses want to go through MR issues they've created at this point and close them (as they will be consolidated in an epic)

>Week 11
>
>**WF** - Review their week issue, just check if there are any big changes that should be flagged up to the MRM

Does this mean that WF will be assigned in week 10?

>Week 12
>
>**MRM** - Should have feedback from all campuses, MRM’s should meet and discuss them. Feedback to the community recommend actions that need to be taken (having considered each campus, any community input and also the learning outcomes)

Is this based on London's timings? At this point in the timeline students would have already started fixing issues/making PRs. Do we want them to start doing this before the MRMs have finished doing their thing? This could lead to students trying to fix a workshop that ends up being stamped as a _major change_ leaving them feeling disempowered as their work it discarded.



### Overall thoughts from timeline
This timetable is assuming that in weeks 10-16 1/2 - 1 full day per week students will be doing curriculum planning. I think that this could be useful to encourage changes to be more punctual, but it's something that I think needs more discussion.

>### Cons
> Changes don’t have to happen. Will have same (softish) deadline for changes (4 weeks before first delivery, apart from the first few weeks off the course).

If significant time is dedicated in weeks 10-16 for curriculum planning, it's feasible to say all changes must happen before week 16. This can avoid deadlines creeping up on students for the later weeks.

**Comment by m4v15** [comment]
>Are you considering reviewing the weekly outcomes before week 10? As we're coming up to the first implementation of this structure it might be worth waiting until post CF feedback to avoid duplicated work.

No probably not, so yeah


> I know this is flagged up in the cons, but this sounds like a huge time commitment.

It's the current process (I think?), and also Im not sure who else can be doing it without taking away the fact that the MRM's will be aware of changes to the curriculum. Also, I think MRM _is_ an important job and will take time. MRM's should be people who have the time to do properly (can be a pre-req for getting the role, dedicating 1-2 hours a week to this)

>Is the idea that each campus makes an epic issue on everything they want changed? Do campuses want to go through MR issues they've created at this point and close them (as they will be consolidated in an epic)

I think probably yeah a big issue for each campus. If the campuses (campi?) want to consolidate previous issues I imagine that would be helpful, but also the MRM's can just look at the other issues when reviewing so not key.

>Does this mean that WF will be assigned in week 10?

WF's can be assigned between now and week 10, MRM's can just do a call out, ask previous WF's

>Is this based on London's timings? At this point in the timeline students would have already started fixing issues/making PRs. Do we want them to start doing this before the MRMs have finished doing their thing? This could lead to students trying to fix a workshop that ends up being stamped as a major change leaving them feeling disempowered as their work it discarded.

I mean I think it's unlikely a lot of things will be wiped out, but also fixing a workshop is a good thing to do even if it stops explicitly being used in the curriculum - having a base of good solid workshops mentors/students could pick from is a really good aim. People can be sensible and fix things that probably won't change too. Also if someone wants to begin a major change, creating a new workshop say, they definitely can whenever, even if it ends up not getting used, again, that's a good thing to have

>This timetable is assuming that in weeks 10-16 1/2 - 1 full day per week students will be doing curriculum planning. I think that this could be useful to encourage changes to be more punctual, but it's something that I think needs more discussion.

It wasn't supposed to be assuming a whole day tbh, just 3-6 hours whenever the CF can find it. It's similar to what happened for FAC11/N/G2 basically. This might be hard in someweeks (week 10 london might be tight) but actually that might be ok if London delay a week and then start in week 11 and then will be doing it the same time as Naz/Gaza. 

Basically I think the timings is something that should be discussed and finalised by the CF's and the MRMs, this is more a starting point.

>If significant time is dedicated in weeks 10-16 for curriculum planning, it's feasible to say all changes must happen before week 16. This can avoid deadlines creeping up on students for the later weeks.
 
Yeah maybe ideally they can all happen by week 16, but if it doesn't happen it's not the end of the world? except for the earlier weeks

**Comment by eliascodes** [comment]
@m4v15 

> WF's can be assigned between now and week 10, MRM's can just do a call out, ask previous WF's

My understanding was that WF's would be assigned reactively based on how extensive the changes to a given week were. That can't happen until the feedback has come in and been reviewed, so week 12/13

**Comment by m4v15** [comment]
@eliascodes Yeah that's true, I was just thinking someone like @ronanyeah might already be happy to be a WF which can just be stated now, even though it might end up that they're not needed. Can then also keep an eye on smaller PR's to their week if that happens before the big feedback.

**Comment by m4v15** [comment]
but ultimately think this isn't _that_ important to have nailed down right now
